sklearn/impute/_base.py

```diff
@@ -72,12 +72,15 @@ def __init__(self, missing_values=np.nan, add_indicator=False):
         self.missing_values = missing_values
         self.add_indicator = add_indicator
 
-    def _fit_indicator(self, X):
+    def _fit_indicator(self, X, mask=None):
         """Fit a MissingIndicator."""
         if self.add_indicator:
+            # if any new missing_values is not passed,
```

Comment:
I am not sure to understand this part. Are you using `missing_values=True` as indicator. However since we will never use the mask, then we will never use the `missing_values` from the `MissingIndicator`.

I think it would be best to not add the `mask` parameter, because at `MissingIndicator().fit(X)`, we are going to inspect `X` to know we if we require `missing_values`.

I would revert this part.

Reply:
It was just to add an extra measure to check whether or not we passed a `precomputed_mask` to `MissingIndicator` internally. There is no harm in removing it as we are using `MissingIndicator` internally, so, we won't we expecting any invalid case.
Sure, I'll remove it in upcoming push.